src/3d/qgspointcloudlayerchunkloader_p.cpp

```diff
@@ -104,6 +104,8 @@ QgsPointCloudLayerChunkLoader::~QgsPointCloudLayerChunkLoader()
   if ( mFutureWatcher && !mFutureWatcher->isFinished() )
   {
     disconnect( mFutureWatcher, &QFutureWatcher<void>::finished, this, &QgsChunkQueueJob::finished );
+    mCanceled = true;
+    mContext.cancelRendering();
```

Comment:
few later below there is cancel() implementation, which does not call `mContext.cancelRendering()` - why is that? It seems like here we should be calling cancel() ?